### PR TITLE
adjusted to core changes

### LINKED ISSUE #6: Update core to v0.2

### PR DESCRIPTION
**Pull request by mat-hek**
closes #6

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,10 +21,10 @@ defmodule FileExamplePipeline do
   def handle_init(_) do
     children = [
       file_src: %File.Source{location: "/etc/passwd"},
-      file_sink: %File.Sink{location: "./test"},
+      file_input: %File.Sink{location: "./test"},
```
Comment (mtsznowak) [inline README.md]: a result of regex?

File: README.md [hunk 2]
```diff
@@ -24,7 +24,7 @@ defmodule FileExamplePipeline do
       file_sink: %File.Sink{location: "./test"},
     ]
     links = %{
-      {:file_src, :source} => {:file_sink, :sink},
+      {:file_src, :source} => {:file_sink, :file_sink},
```
Comment (bblaszkow06) [inline README.md]: LOL, regex again?

File: lib/membrane_element_file/sink.ex [hunk 3]
```diff
@@ -25,23 +25,24 @@ defmodule Membrane.Element.File.Sink do
...
-  def handle_play(_, state) do
-    {{:ok, demand: :sink}, state}
+  def handle_prepared_to_playing(_ctx, state) do
+    {{:ok, demand: :input}, state}
   end
 
   @impl true
-  def handle_write1(:sink, %Buffer{payload: payload}, _, %{fd: fd} = state) do
+  def handle_write(:input, %Buffer{payload: payload}, _ctx, %{fd: fd} = state) do
+    :timer.sleep(30)
```
Comment (bblaszkow06) [inline lib/membrane_element_file/sink.ex]: Umm is this some debugging leftover or am I missing something?

File: lib/membrane_element_file/sink_multi.ex [hunk 4]
```diff
@@ -38,15 +38,15 @@ defmodule Membrane.Element.File.Sink.Multi do
                 path/to/file0.ext, path/to/file1.ext, ...
                 """
               ],
-              split_event_type: [
-                type: :atom,
-                default: :split,
-                description: "Type of event causing switching to a new file"
+              split_event: [
+                type: :struct,
```
Comment (bblaszkow06) [inline lib/membrane_element_file/sink_multi.ex]: I don't think its a good idea to pass struct instance here. It's safer to pass struct module name. Imagine a situation where we'd like to use non-empty struct as `split_event`. The options would have to contain some values for enforced keys and that's not something we want

File: lib/membrane_element_file/sink_multi/split.ex [hunk 5]
```diff
@@ -0,0 +1,9 @@
+defmodule Membrane.Element.File.Sink.Multi.Split do
```
Comment (bblaszkow06) [inline lib/membrane_element_file/sink_multi/split.ex]: That name is horribly long. How about renaming it to `Membrane.Element.File.SplitEvent` and moving to proper catalogue?

File: mix.exs [hunk 6]
```diff
@@ -49,7 +49,8 @@ defmodule Membrane.Element.File.Mixfile do
   defp deps do
     [
       {:ex_doc, "~> 0.19", only: :dev, runtime: false},
-      {:membrane_core, git: "https://github.com/membraneframework/membrane-core.git"},
+      {:membrane_core, github: "membraneframework/membrane-core", branch: "pre-review"},
+      {:bunch, github: "membraneframework/bunch", branch: "feature/default_access"},
```
Comment (bblaszkow06) [inline mix.exs]: The deps here can be changed (to master for both deps)

**Comment by bblaszkow06** [comment]
BTW, have you seen this? https://github.com/membraneframework/membrane-element-file/compare/feature/release_v0.2. You can cherry-pick travis fix

File: lib/membrane_element_file/split.ex [hunk 7]
```diff
@@ -1,4 +1,4 @@
-defmodule Membrane.Element.File.Sink.Multi.Split do
+defmodule Membrane.Element.File.SplitEvent do
```
Comment (bblaszkow06) [inline lib/membrane_element_file/split.ex]: Please rename the file accordingly

**Pull request merged**